### PR TITLE
Cleanup deprecation warning for c++17 found in vs2019

### PR DESCRIPTION
**Pull request by bpmckinnon**
I was getting a lot of warnings when compiling by code so I added some checks that should also help prepare for the removal of these functions in c++20

### CODE REVIEW

File: parallel_hashmap/btree.h [hunk 1]
```diff
@@ -712,7 +712,7 @@ namespace phmap {
 
         template <
             typename Compare, typename K, typename LK,
-            phmap::enable_if_t<!std::is_same<bool, phmap::result_of_t<
+            phmap::enable_if_t<!std::is_same<bool, phmap::invoke_result_t<
                                                        Compare(const K &, const LK &)>>::value,
```
Comment (greg7mdp) [inline parallel_hashmap/btree.h:716]: Compare, const K &, const LK &>>::value,

File: parallel_hashmap/btree.h [hunk 2]
```diff
@@ -721,7 +721,7 @@ namespace phmap {
         }
         template <
             typename Compare, typename K, typename LK,
-            phmap::enable_if_t<std::is_same<bool, phmap::result_of_t<Compare(
+            phmap::enable_if_t<std::is_same<bool, phmap::invoke_result_t<Compare(
```
Comment (greg7mdp) [inline parallel_hashmap/btree.h:724]: same here

File: parallel_hashmap/btree.h (hunk 1 above)
Comment (bpmckinnon) [reply parallel_hashmap/btree.h:716]: I had it right at one point, but I put myself into some kind of funk.  I'm still working on my git skills.

**Comment by greg7mdp** [comment]
Awesome, thanks again for this brian!

**Comment by bpmckinnon** [comment]
No problem.  I'll keep an eye out for anything else while I'm working with the library!

**Comment by greg7mdp** [comment]
Sounds good, thanks!

**Comment by kraliq233** [comment]
Hs

**Pull request merged**